dataland-frontend/src/components/resources/frameworkDataSearch/lksg/LksgCompanyDataTable.vue

```diff
@@ -41,28 +41,28 @@
         :key="reportingPeriod.dataId"
       >
         <template #body="{ data }">
-          <a
-            v-if="
-              Array.isArray(data[reportingPeriod.dataId]) &&
-              (data[reportingPeriod.dataId].length > 1 ||
-                data[reportingPeriod.dataId].some((el) => typeof el === 'object'))
-            "
-            @click="openModalAndDisplayValuesInSubTable(data[reportingPeriod.dataId], data.kpiLabel, data.kpiKey)"
-            class="link"
-            >Show "{{ data.kpiLabel }}"
-            <em class="material-icons" aria-hidden="true" title=""> dataset </em>
-          </a>
-          <span v-else-if="typeof data[reportingPeriod.dataId] === 'object' && data[reportingPeriod.dataId]?.value">
-            {{ data[reportingPeriod.dataId].value }}
-          </span>
-
-          <span v-else
-            >{{
-              Array.isArray(data[reportingPeriod.dataId])
-                ? data[reportingPeriod.dataId][0]
-                : data[reportingPeriod.dataId]
-            }}
-          </span>
+          <template v-if="data[reportingPeriod.dataId] !== undefined && data[reportingPeriod.dataId] !== null">
+            <template v-if="Array.isArray(data[reportingPeriod.dataId])">
+              <a
+                v-if="
+                  data[reportingPeriod.dataId].length > 1 ||
+                  data[reportingPeriod.dataId].some((el) => typeof el === 'object')
+                "
+                @click="openModalAndDisplayValuesInSubTable(data[reportingPeriod.dataId], data.kpiLabel, data.kpiKey)"
+                class="link"
+                >Show "{{ data.kpiLabel }}"
+                <em class="material-icons" aria-hidden="true" title=""> dataset </em>
+              </a>
+              <span v-else> data[reportingPeriod.dataId][0] </span>
+            </template>
+            <span v-else-if="data.kpiFormFieldComponent == 'PercentageFormField'">
+              {{ data[reportingPeriod.dataId] }} %</span
+            >
+            <span v-else-if="typeof data[reportingPeriod.dataId] === 'object' && data[reportingPeriod.dataId]?.value">
+              {{ data[reportingPeriod.dataId].value }}
+            </span>
+            <span v-else>{{ data[reportingPeriod.dataId] }} </span>
+          </template>
         </template>
       </Column>
 
```

Comment:
After fixing this, also the "v-for" in the second column should do smth like 
`v-for="columnIdentifier of columnIdentifierObjects"

Reply:
I have refactored the very generic `columnIdentifierObjects` it with the introduction of a type `DataSetReportingPeriod` that is more specific and makes more sense here in my opinion. The same goes for the following reviews that deal with this variable.

Reply:
Having a type for the column identifier objects is nice, I like that.

But the naming of that type with `DataSetReportingPeriod` is the opposite of what I meant:

I actually liked the idea of calling it `ColumnIdentifierObject` or something similar.
`DataSetReportinPeriod` is misleading because it implies that it only bears a reporting period, while it also bears the data Id and serves the sole purpose of identifying the different columns.

The current naming leads to code which is strange to read, like here:

![image](https://github.com/d-fine/Dataland/assets/99255296/45cff1c4-a300-4433-bef0-3e5b2866035b)

The first time I read this I was really confused.

Reply:
The reason I did not want to name it `columnIdentifierObject` is because I find it very confusing when such an abstract object has a very concrete field named `reportingPeriod`, there is no logical connection between the object name and the field name. So I would go with your second suggestion `DataSetReportingPeriodAndDataId` . Additionally I would refactor this loop to work like `v-for="(reportingPeriod, dataId) of dataSetReportingPeriodAndDataId"`. This is a much clearer way to extract this tuple, is that alright for you?

Reply:
Sounds great, thx !

Reply:
Yes it has, thank you.  There's only one thing that still needs to be adjusted:

It is the prop name of the array, which holds the `DataSetReportingPeriodAndDataId`-objects.
It is still named in a way, that one might think it only holds reporting periods:

![image](https://github.com/d-fine/Dataland/assets/99255296/e5ce59e8-1eac-4e1a-a757-6499a3585dc1)

Reply:
And why have you added an index signature to `ReportingPeriodOfDataSetWithId`?

![image](https://github.com/d-fine/Dataland/assets/99255296/d41e4ac7-6da2-4506-98b6-730148e49bc0)

Reply:
Perfect. Closing this.